[16.0][MIG] account_fiscal_position_type (from 12.0 to 16.0)

legalsylvain opened the pull request:
### Description

This module extends the functionality of account to add a use type on fiscal position. (``sale``, ``purchase`` or ``all``) to restrict the usage of fiscal position for out or in invoices.

* If a fiscal position is configured for sale, it will not be possible to use it on a customer invoice

* If a fiscal position is configured for purchase, it will not be possible to use it on a supplier invoice

### Configure

To configure this module, you need to:

* Go to Invoicing / Configuration / Taxes / Fiscal Positions

* On a fiscal position, set the correct settings.

![fiscal_position_form](https://user-images.githubusercontent.com/3407482/196276383-6bb0cec6-b2d5-4525-8f3d-758937abe467.png)


**Note**

It is possible to set the value on the fiscal position templates. In that case, installing a new chart of account will configure correctly the fiscal positions.

legalsylvain committed:
[ADD][V8.0] account_fiscal_position_type

legalsylvain committed:
[MIG][12.0] Port account_fiscal_position_type

quentinDupont committed:
[FIX] wrong attribute

oca-travis committed:
[UPD] Update account_fiscal_position_type.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
account_fiscal_position_type 12.0.1.0.1

legalsylvain committed:
[FIX] account_fiscal_position_type : do not crash if domain is not set

OCA-git-bot committed:
account_fiscal_position_type 12.0.1.0.2

quentinDupont committed:
Added translation using Weblate (French)

quentinDupont committed:
Translated using Weblate (French)

Currently translated at 63.6% (7 of 11 strings)

Translation: account-fiscal-rule-12.0/account-fiscal-rule-12.0-account_fiscal_position_type
Translate-URL: https://translation.odoo-community.org/projects/account-fiscal-rule-12-0/account-fiscal-rule-12-0-account_fiscal_position_type/fr/

legalsylvain committed:
[IMP] account_fiscal_position_type: black, isort, prettier

legalsylvain committed:
[MIG] port account_fiscal_position_type from 12.0 to 16.0
- remove check on res.partner, as the field 'customer' and 'supplier' disappeared
- add new demo chart of account to avoid to depends on l10n_generic_coa that makes test failing in some configuration
- refactor the algorithm, introducing suitable_fiscal_position_ids field (as Odoo does with suitable_journal_ids field)
- Update documentation

rafaelbn commented:
/ocabot migration account_fiscal_position_type

rafaelbn commented:
Hello @legalsylvain ,

Tested and it works. I think this is the opposite

>     If a fiscal position is configured for sale, it will not be possible to use it on a customer invoice
If a fiscal position is configured for sale, it will not be possible to use it on a vendor bill
>     If a fiscal position is configured for purchase, it will not be possible to use it on a supplier invoice
If a fiscal position is configured for purchase, it will not be possible to use it on a customer invoice

![imagen](https://user-images.githubusercontent.com/8736623/196425198-b719ed65-164f-4da4-931e-777cf5e136b9.png)

rafaelbn reviewed (commented):
Tested, just IMHO fix readme 😄 👍

rafaelbn commented:
In my dreams you can set for `res.partner` diferent fiscal position for sales and purchases 😄

rafaelbn reviewed (approved):
README is correct! 👍 

* If a fiscal position is configured for sale, it will not be possible to use
  it in a purchase invoice.

* If a fiscal position is configured for sale, it will not be possible to
  set it to a partner flagged only as 'supplier'

rafaelbn commented on account_fiscal_position_type/README.rst:
```diff
@@ -0,0 +1,94 @@
+usage of fiscal position for out or in invoices.
+
+* If a fiscal position is configured for sale, it will not be possible to use
+  it in a purchase invoice.
```
Odoo terminology is "Vendor bill". Just details.

legalsylvain commented:
Hi @rafaelbn. Thanks for your review !
- I changed the text for that one : 

```
This module extends the functionality of account to add a use type on
fiscal position. (``sale``, ``purchase`` or ``all``) to restrict the
usage of fiscal position for out or in invoices.

* If a fiscal position is configured for sale, it will not be possible to use it
  on Vendor Bills, Refunds and Purchase Receipts.

* If a fiscal position is configured for purchase, it will not be possible to use it
  on Customer Invoices, Credit Notes and Sales Receipts.

```

- also I added comment in the commit text.

thanks !

Luisotto reviewed (approved):
Tested! LGTM!

pedrobaeza commented:
/ocabot merge nobump

SoniaViciana commented:
Hi @dreispt,

This module is merged, however the translations are locked:

[account-fiscal-rule-16.0/account-fiscal-rule-16.0-account_fiscal_position_type](https://translation.odoo-community.org/projects/account-fiscal-rule-16-0/account-fiscal-rule-16-0-account_fiscal_position_type/)

Can I know why?

Thanks in advance,

kikopeiro commented:
> Hi @dreispt,
> 
> This module is merged, however the translations are locked:
> 
> [account-fiscal-rule-16.0/account-fiscal-rule-16.0-account_fiscal_position_type](https://translation.odoo-community.org/projects/account-fiscal-rule-16-0/account-fiscal-rule-16-0-account_fiscal_position_type/)
> 
> Can I know why?
> 
> Thanks in advance,

CC @rafaelbn
